src/main/java/apaintus/controllers/CanvasController.java

```diff
@@ -160,7 +176,40 @@ private void saveDrawLayer() {
             canvasService.clear(drawLayer.getGraphicsContext2D());
         }
     }
-    
+
+    private void saveSelectionBox() {
+        if (activeShape != null && !drawnShapes.contains(activeShape)) {
+            for (DrawableShape shape : drawnShapes) {
+                if (selectionBox.contains(shape)) {
+                    selectionBox.add(shape);
+                }
+            }
+
+            if (selectionBox.getSize() > 1) {
+                selectionBox.resize();
+
+                for (DrawableShape shape : drawnShapes) {
+                    if (shape.getShapeType() == ShapeType.SELECTION_BOX) {
+                        activeShape = selectionBox.isDuplicate(shape) ? shape : selectionBox;
+                        activeShape.setSelected(true);
+                    }
+                }
+
+                if (activeShape == selectionBox) {
+                    selectionBox.optimize();
+                    drawnShapes.add(activeShape);
+                }
+            }
+        }
+
+        if (!drawnShapes.contains(activeShape)) {
+            attributeController.resetAttributes();
+        }
+
+        canvasService.clear(drawLayer.getGraphicsContext2D());
```

Comment:
save selection box shouldn't have to redraw the canvas??

Reply:
En fait, puisque la selection box est resize pour fit les shapes à l'intérieur, il faut provoquer un redraw. Sinon, si on apply comme dans saveState, la bounding box de la selection box sera de la taille du user